### PR TITLE
Use more precise last-modified times for files

### PR DESCRIPTION
**Pull request by AA-Turner**

### CODE REVIEW

File: sphinx/util/osutil.py [hunk 1]
```diff
@@ -72,20 +71,25 @@ def ensuredir(file: str | os.PathLike[str]) -> None:
     os.makedirs(file, exist_ok=True)
 
 
-def mtimes_of_files(dirnames: list[str], suffix: str) -> Iterator[float]:
```
Comment (picnixz) [inline sphinx/util/osutil.py:75]: While those were never documented, to be on the safe side, let's make them follow a deprecation period.

File: sphinx/util/osutil.py [hunk 2]
```diff
@@ -72,20 +71,25 @@ def ensuredir(file: str | os.PathLike[str]) -> None:
...
-                    with contextlib.suppress(OSError):
-                        yield path.getmtime(path.join(root, sfile))
+def _last_modified_time(source: str | os.PathLike[str], /) -> int:
+    """Return the last modified time of ``filename``.
 
+    The time is returned as integer microseconds.
+    The lowest common denominator of modern file-systems seems to be
+    microsecond-level precision.
 
-def copytimes(source: str | os.PathLike[str], dest: str | os.PathLike[str]) -> None:
```
Comment (picnixz) [inline sphinx/util/osutil.py:84]: ditto

**Review by picnixz — COMMENTED** [review]
I don't know how this would affect the tests I patched for Windows (hopefully the mocks are fine but please tell me whether you checked them before)

File: sphinx/util/osutil.py (hunk 1 above)
Comment (AA-Turner) [reply sphinx/util/osutil.py:75]: There are no uses in the wild: https://github.com/search?q=mtimes_of_files%28+NOT+is%3Aarchived+NOT+is%3Afork+language%3APython&type=code

File: sphinx/util/osutil.py (hunk 2 above)
Comment (AA-Turner) [reply sphinx/util/osutil.py:84]: Likewise: https://github.com/search?q=copytimes%28+NOT+is%3Aarchived+NOT+is%3Afork+language%3APython&type=code&p=1

File: sphinx/util/osutil.py (hunk 1 above)
Comment (picnixz) [reply sphinx/util/osutil.py:75]: Oh then it's fine I think.

**Pull request merged**